bundles/EcommerceFrameworkBundle/IndexService/ProductList/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -850,9 +887,27 @@ protected function buildQueryConditions($queryFilters, $excludedFieldnames)
      */
     protected function loadElementById($elementId)
     {
-        return $this->tenantConfig->getObjectMockupById($elementId);
+        /** @var ElasticSearch $tenantConfig */
+        $tenantConfig = $this->getTenantConfig();
+        if ($this->isLoadFromSource()) {
+            $mockup = null;
+            if (isset($this->hitData[$elementId])) {
```

Comment:
Do you have a specific idea in mind?
If we remove the caching feature, then it always needs to be a direct mapping, or are you thinking about some kind of static cache so that the mapping is only done once per (URL) request?

I am not 100% sure if I get your point?